src/linear_algebra/basis.lean

```diff
@@ -343,6 +343,21 @@ end
 
 end is_basis
 
+lemma is_basis_singleton_iff {R : Type*} [division_ring R] [module R M]
```

Comment:
This holds in integral domains as well (if `M` has no torsion). This is the highest generality I could get without needing to change the proof:
```suggestion
lemma is_basis_singleton_iff {R : Type*} [ring R] [nontrivial R] [module R M] [no_zero_smul_divisors R M]
```
where we need to generalize the following lemmas in `linear_independent.lean`:
```lean
lemma linear_independent_unique_iff [ring R] [nontrivial R] [add_comm_monoid M] [module R M]
  [no_zero_smul_divisors R M]
  (v : ι → M) [unique ι] :
  linear_independent R v ↔ v (default ι) ≠ 0 :=
begin
  simp only [linear_independent_iff, finsupp.total_unique, smul_eq_zero],
  refine ⟨λ h hv, _, λ hv l hl, finsupp.unique_ext $ hl.resolve_right hv⟩,
  have := h (finsupp.single (default ι) 1) (or.inr hv),
  exact one_ne_zero (finsupp.single_eq_zero.1 this)
end

alias linear_independent_unique_iff ↔ _ linear_independent_unique

lemma linear_independent_singleton {x : V} (hx : x ≠ 0) :
  linear_independent K (λ x, x : ({x} : set V) → V) :=
linear_independent_unique coe hx
```

Reply:
Thanks for these improvements!